Update Stats page to allow easy download of graphs

rileyhgrant opened the pull request:
Resolves #1265 

Adds a button to allow easy downloads of graphs on the stats page

mattsolo1 commented:
Works as expected!

The download button is huge, and most visitors will not want to download anything. What do you think about adding a more subtle icon? We already have an icon package installed (font awesome). Here are some options: https://fontawesome.com/search?q=download&o=r&m=free. I was thinking maybe it would fit in the legend or next to the title (see below)?

A few other general points about the plot (not related to downloading). @sambaxter do you have feedback?

- Normally people expect graphs to have titles. Perhaps something like "Sample size across major ExAC/gnomAD releases"
- Would be cool if the year of each was release was included below the release title to emphasize the long history of the project (2014, 2016, 2019, 2023)
- There is no Y axis label... should it be "Number of samples"?
- There is a point where the text overlaps:

![image](https://github.com/broadinstitute/gnomad-browser/assets/8867295/49cece32-bd3a-4d9f-b6c5-11bed3a1dbca)

Could set the minimum width of this plot to be larger?

rileyhgrant commented:
Thanks for the feedback, the button is pretty huge and imposing. I'll make it more subtle and incorporate your other suggestions.

rileyhgrant commented:
Updated with:
- Smaller download button with icon (next to title)
- Added prop for title so all bar graphs have a title
  - id of element and download are now generated by slugifying the title of the graph
- Cleaning up: defined y axis for graphs, used same small download icon on tables, added titles to all graphs

![Screenshot 2024-01-02 at 15 06 07](https://github.com/broadinstitute/gnomad-browser/assets/59549713/dda288c4-2037-4931-bdcd-7b5bb983abeb)
